### PR TITLE
Feature/7323 change user

### LINKED ISSUE #7323: Add "changeUser" field to entities and history to know which user last changed an entity [5]
### Situation Description
Currently we only have the information on when an entity was created and changed and which user reported it. What is missing is the information who changed an entity. This can currently only be extracted from the audit database. 

In addition there is currently some redundancy between the audit database and the history tables that we want to get rid of. Our idea is to use the history tables to keep track of all changes made to entities (create, update, delete) and to provide a separate log that provides the information that a user has changed or accessed a resource without containing any actual data on the resource itself, so that we don't have any person related data in there.

### Feature Description

- [x] Add a "changeUser" field to the AbstractDomainObject and to all related database tables and history tables
- [x] update the changeUser of the AbstractDomainObject using the PrePersist and PreUpdate annotation. See User.updateJursidictionLevel

**Important**: We have to validate whether this approach actually works and only set's the user when the entity was changed. 
Otherwise we may have to detect this somehow or work with an interceptor. In this case we should have a look at the mechanics of the existing audit log.

### Additional Information

- We recently had the approach to touch all `AbstractDomainObject`s in SQL with a foreach loop: https://github.com/hzi-braunschweig/SORMAS-Project/issues/7303#issuecomment-1015185867
- For now we should not include this in the DTO classes and especially don't have to include it in the mobile app sync process. This is a backend only functionality.

### PR DESCRIPTION
**Pull request by JonasCir**
Fixes #7323 

@MartinWahnschaffe I contacted @HolgerReiseVSys, but the only working solution I came up with is to give `getCurrentUser` its own transaction. Nothing else worked for me.

### CODE REVIEW

**Comment by JonasCir** [comment]
We definitely should run performance tests and see for the impact.

File: sormas-backend/src/main/resources/sql/sormas_schema.sql [hunk 1]
```diff
@@ -9798,4 +9798,607 @@ ALTER TABLE weeklyreportentry_history ALTER COLUMN creationdate TYPE timestamp(3
...
 
+-- 2022-01-22 change by user #7323
+
+ALTER TABLE action ADD COLUMN change_user_id BIGINT,
+                   ADD CONSTRAINT fk_change_user_id
+                       FOREIGN KEY (change_user_id)
+                           REFERENCES users (id);
+
+ALTER TABLE action_history ADD COLUMN change_user_id BIGINT,
+                           ADD CONSTRAINT fk_change_user_id
```
Comment (MartinWahnschaffe) [inline sormas-backend/src/main/resources/sql/sormas_schema.sql:9809]: @JonasCir The history tables in general should not have foreign keys.

**Comment by JonasCir** [comment]
This approach also nukes the tests, as the bean manager does not seem to be available there.

File: sormas-backend/src/main/resources/sql/sormas_schema.sql (hunk 1 above)
Comment (JonasCir) [reply sormas-backend/src/main/resources/sql/sormas_schema.sql]: fixed

**Comment by MartinWahnschaffe** [comment]
@JonasCir Looks like some tests fail with the following error.
I was able to reproduce this on my local system (e.g. CaseImporterTest.testLineListingImport).

> Caused by: javax.naming.NoInitialContextException: Need to specify class name in environment or system property, or in an application resource file: java.naming.factory.initial
> 	at java.naming/javax.naming.spi.NamingManager.getInitialContext(NamingManager.java:702)
> 	at java.naming/javax.naming.InitialContext.getDefaultInitCtx(InitialContext.java:305)
> 	at java.naming/javax.naming.InitialContext.getURLOrDefaultInitCtx(InitialContext.java:342)
> 	at java.naming/javax.naming.InitialContext.lookup(InitialContext.java:409)
> 	at de.symeda.sormas.backend.common.AbstractDomainObject$AdoListener.getCurrentUser(AbstractDomainObject.java:188)

2022-03-11T16:37:36.7473713Z [ERROR]   CaseImporterTest>AbstractBeanTest.init:99 » Runtime javax.naming.NoInitialCont...
2022-03-11T16:37:36.7475335Z [ERROR]   CaseImporterTest>AbstractBeanTest.init:99 » Runtime javax.naming.NoInitialCont...
2022-03-11T16:37:36.7476909Z [ERROR]   CaseImporterTest>AbstractBeanTest.init:99 » Runtime javax.naming.NoInitialCont...
2022-03-11T16:37:36.7478469Z [ERROR]   CaseImporterTest>AbstractBeanTest.init:99 » Runtime javax.naming.NoInitialCont...
2022-03-11T16:37:36.7479910Z [ERROR]   CaseImporterTest>AbstractBeanTest.init:99 » Runtime javax.naming.NoInitialCont...
2022-03-11T16:37:36.7481321Z [ERROR]   CaseImporterTest>AbstractBeanTest.init:99 » Runtime javax.naming.NoInitialCont...
2022-03-11T16:37:36.7483203Z [ERROR]   ContactImporterTest>AbstractBeanTest.init:99 » Runtime javax.naming.NoInitialC...
2022-03-11T16:37:36.7484529Z [ERROR]   ContactImporterTest>AbstractBeanTest.init:99 » Runtime javax.naming.NoInitialC...
2022-03-11T16:37:36.7485846Z [ERROR]   ContactImporterTest>AbstractBeanTest.init:99 » Runtime javax.naming.NoInitialC...
2022-03-11T16:37:36.7487169Z [ERROR]   CampaignFormDataImporterTest>AbstractBeanTest.init:99 » Runtime javax.naming.N...
2022-03-11T16:37:36.7488519Z [ERROR]   EventParticipantImporterTest>AbstractBeanTest.init:99 » Runtime javax.naming.N...
2022-03-11T16:37:36.7489774Z [ERROR]   EventParticipantImporterTest>AbstractBeanTest.init:99 » Runtime javax.naming.N...
2022-03-11T16:37:36.7491114Z [ERROR]   EventParticipantImporterTest>AbstractBeanTest.init:99 » Runtime javax.naming.N...
2022-03-11T16:37:36.7491907Z [ERROR]   EventParticipantImporterTest>AbstractBeanTest.init:99 » Runtime javax.naming.N...
2022-03-11T16:37:36.7492669Z [ERROR]   EventParticipantImporterTest>AbstractBeanTest.init:99 » Runtime javax.naming.N...
2022-03-11T16:37:36.7493418Z [ERROR]   EventParticipantImporterTest>AbstractBeanTest.init:99 » Runtime javax.naming.N...
2022-03-11T16:37:36.7494164Z [ERROR]   EventParticipantImporterTest>AbstractBeanTest.init:99 » Runtime javax.naming.N...
2022-03-11T16:37:36.7494906Z [ERROR]   EventImporterTest>AbstractBeanTest.init:99 » Runtime javax.naming.NoInitialCon...
2022-03-11T16:37:36.7495651Z [ERROR]   CountryImporterTest>AbstractBeanTest.init:99 » Runtime javax.naming.NoInitialC...
2022-03-11T16:37:36.7496390Z [ERROR]   CountryImporterTest>AbstractBeanTest.init:99 » Runtime javax.naming.NoInitialC...
2022-03-11T16:37:36.7497138Z [ERROR]   InfrastructureImporterTest>AbstractBeanTest.init:99 » Runtime javax.naming.NoI...
2022-03-11T16:37:36.7497902Z [ERROR]   InfrastructureImporterTest>AbstractBeanTest.init:99 » Runtime javax.naming.NoI...
2022-03-11T16:37:36.7498661Z [ERROR]   InfrastructureImporterTest>AbstractBeanTest.init:99 » Runtime javax.naming.NoI...
2022-03-11T16:37:36.7499412Z [ERROR]   InfrastructureImporterTest>AbstractBeanTest.init:99 » Runtime javax.naming.NoI...
2022-03-11T16:37:36.7500145Z [ERROR]   InfrastructureImporterTest>AbstractBeanTest.init:99 » Runtime javax.naming.NoI...
2022-03-11T16:37:36.7500884Z [ERROR]   InfrastructureImporterTest>AbstractBeanTest.init:99 » Runtime javax.naming.NoI...
2022-03-11T16:37:36.7501623Z [ERROR]   InfrastructureImporterTest>AbstractBeanTest.init:99 » Runtime javax.naming.NoI...
2022-03-11T16:37:36.7502356Z [ERROR]   InfrastructureImporterTest>AbstractBeanTest.init:99 » Runtime javax.naming.NoI...
2022-03-11T16:37:36.7503094Z [ERROR]   ImportExportTest>AbstractBeanTest.init:99 » Runtime javax.naming.NoInitialCont...
2022-03-11T16:37:36.7503844Z [ERROR]   ImportExportTest>AbstractBeanTest.init:99 » Runtime javax.naming.NoInitialCont...
2022-03-11T16:37:36.7504771Z [ERROR]   LabMessageMapperTest>AbstractBeanTest.init:99 » Runtime javax.naming.NoInitial...
2022-03-11T16:37:36.7505621Z [ERROR]   LabMessageMapperTest>AbstractBeanTest.init:99 » Runtime javax.naming.NoInitial...
2022-03-11T16:37:36.7506376Z [ERROR]   LabMessageMapperTest>AbstractBeanTest.init:99 » Runtime javax.naming.NoInitial...
2022-03-11T16:37:36.7507124Z [ERROR]   LabMessageMapperTest>AbstractBeanTest.init:99 » Runtime javax.naming.NoInitial...
2022-03-11T16:37:36.7507872Z [ERROR]   RelatedLabMessageHandlerTest.init:95->AbstractBeanTest.init:99 » Runtime javax...
2022-03-11T16:37:36.7508646Z [ERROR]   RelatedLabMessageHandlerTest.init:95->AbstractBeanTest.init:99 » Runtime javax...
2022-03-11T16:37:36.7509411Z [ERROR]   RelatedLabMessageHandlerTest.init:95->AbstractBeanTest.init:99 » Runtime javax...
2022-03-11T16:37:36.7510160Z [ERROR]   RelatedLabMessageHandlerTest.init:95->AbstractBeanTest.init:99 » Runtime javax...
2022-03-11T16:37:36.7510928Z [ERROR]   RelatedLabMessageHandlerTest.init:95->AbstractBeanTest.init:99 » Runtime javax...
2022-03-11T16:37:36.7511687Z [ERROR]   RelatedLabMessageHandlerTest.init:95->AbstractBeanTest.init:99 » Runtime javax...
2022-03-11T16:37:36.7512561Z [ERROR]   RelatedLabMessageHandlerTest.init:95->AbstractBeanTest.init:99 » Runtime javax...
2022-03-11T16:37:36.7513313Z [ERROR]   RelatedLabMessageHandlerTest.init:95->AbstractBeanTest.init:99 » Runtime javax...
2022-03-11T16:37:36.7514060Z [ERROR]   RelatedLabMessageHandlerTest.init:95->AbstractBeanTest.init:99 » Runtime javax...
2022-03-11T16:37:36.7514915Z [ERROR]   RelatedLabMessageHandlerTest.init:95->AbstractBeanTest.init:99 » Runtime javax...
2022-03-11T16:37:36.7515658Z [ERROR]   RelatedLabMessageHandlerTest.init:95->AbstractBeanTest.init:99 » Runtime javax...
2022-03-11T16:37:36.7516404Z [ERROR]   RelatedLabMessageHandlerTest.init:95->AbstractBeanTest.init:99 » Runtime javax...
2022-03-11T16:37:36.7517170Z [ERROR]   RelatedLabMessageHandlerTest.init:95->AbstractBeanTest.init:99 » Runtime javax...
2022-03-11T16:37:36.7517920Z [ERROR]   RelatedLabMessageHandlerTest.init:95->AbstractBeanTest.init:99 » Runtime javax...
2022-03-11T16:37:36.7518668Z [ERROR]   RelatedLabMessageHandlerTest.init:95->AbstractBeanTest.init:99 » Runtime javax...
2022-03-11T16:37:36.7519413Z [ERROR]   RelatedLabMessageHandlerTest.init:95->AbstractBeanTest.init:99 » Runtime javax...
2022-03-11T16:37:36.7520137Z [ERROR]   RelatedLabMessageHandlerTest.init:95->AbstractBeanTest.init:99 » Runtime javax...
2022-03-11T16:37:36.7520908Z [ERROR]   RelatedLabMessageHandlerTest.init:95->AbstractBeanTest.init:99 » Runtime javax...
2022-03-11T16:37:36.7521652Z [ERROR]   RelatedLabMessageHandlerTest.init:95->AbstractBeanTest.init:99 » Runtime javax...
2022-03-11T16:37:36.7522396Z [ERROR]   RelatedLabMessageHandlerTest.init:95->AbstractBeanTest.init:99 » Runtime javax...
2022-03-11T16:37:36.7523144Z [ERROR]   RelatedLabMessageHandlerTest.init:95->AbstractBeanTest.init:99 » Runtime javax...
2022-03-11T16:37:36.7523891Z [ERROR]   RelatedLabMessageHandlerTest.init:95->AbstractBeanTest.init:99 » Runtime javax...
2022-03-11T16:37:36.7524641Z [ERROR]   RelatedLabMessageHandlerTest.init:95->AbstractBeanTest.init:99 » Runtime javax...
2022-03-11T16:37:36.7525378Z [ERROR]   RelatedLabMessageHandlerTest.init:95->AbstractBeanTest.init:99 » Runtime javax...
2022-03-11T16:37:36.7526124Z [ERROR]   RelatedLabMessageHandlerTest.init:95->AbstractBeanTest.init:99 » Runtime javax...
2022-03-11T16:37:36.7526868Z [ERROR]   RelatedLabMessageHandlerTest.init:95->AbstractBeanTest.init:99 » Runtime javax...
2022-03-11T16:37:36.7527606Z [ERROR]   RelatedLabMessageHandlerTest.init:95->AbstractBeanTest.init:99 » Runtime javax...
2022-03-11T16:37:36.7528345Z [ERROR]   SampleControllerTest>AbstractBeanTest.init:99 » Runtime javax.naming.NoInitial...
2022-03-11T16:37:36.7529090Z [ERROR]   DownloadUtilTest>AbstractBeanTest.init:99 » Runtime javax.naming.NoInitialCont...
2022-03-11T16:37:36.7529837Z [ERROR]   TravelEntryImporterTest>AbstractBeanTest.init:99 » Runtime javax.naming.NoInit...
2022-03-11T16:37:36.7531022Z [ERROR]   TravelEntryImporterTest>AbstractBeanTest.init:99 » Runtime javax.naming.NoInit...
2022-03-11T16:37:36.7531788Z [ERROR]   TravelEntryImporterTest>AbstractBeanTest.init:99 » Runtime javax.naming.NoInit...
2022-03-11T16:37:36.7532531Z [ERROR]   TravelEntryImporterTest>AbstractBeanTest.init:99 » Runtime javax.naming.NoInit...
2022-03-11T16:37:36.7533264Z [ERROR]   TravelEntryImporterTest>AbstractBeanTest.init:99 » Runtime javax.naming.NoInit...

**Review by MartinWahnschaffe — APPROVED** [review]
Besides the failed tests, the implementation looks good.
Can be merged, once the github CI gives green light.

**Pull request merged**